include/souper/Inst/Inst.h

```diff
@@ -116,6 +116,9 @@ struct Inst : llvm::FoldingSetNode {
   static bool isAssociative(Kind K);
   static bool isCommutative(Kind K);
   static int getCost(Kind K);
+  unsigned KnownZeros;
+  unsigned KnownOnes;
+  std::string BitPattern;
```

Comment:
is there a reason to add this string to Inst? why not just create it when needed?